buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -179,6 +179,16 @@ class BeamModulePlugin implements Plugin<Project> {
      * The set of additional maven repositories that should be added into published POM file.
      */
     List<Map> mavenRepositories = []
+
+    /**
+     * Flag to compile the main source with testJavaVersion project property (if set).
+     *
+     * Set this configuration if the project sources are supposed to be compiled in given Java version
+     * for testing propose (e.g. pipeline Examples).
+     *
+     * <p>If testJavaVersion project property is not set (e.g. publish artifacts), it should has no effect.
+     */
+    boolean compileWithTestJavaVersion = false
```

Comment:
We need a configuration for plugin instead of project property because this is project (e.g. `:examples:java`) specific. During a build only test fixture project should be compiled with testJavaVersion